sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -729,7 +729,7 @@ class DataSourceV2SQLSuite
     val ex = intercept[AnalysisException] {
       sql("DROP TABLE testcat.db.notbl")
     }
-    assert(ex.getMessage.contains("Table or view not found: testcat.db.notbl"))
+    assert(ex.getMessage.contains("Table or view not found for 'DROP TABLE': testcat.db.notbl"))
```

Comment:
This message could be misleading since `DROP TABLE` supports table and a temporary view. To fix this, we can add `allowPermanentView` as the following and update `Analyzer` accordingly:
```scala
case class UnresolvedTableOrView(
    multipartIdentifier: Seq[String],
    commandName: String,
    allowTempView: Boolean = true,
    allowPermanentView: Boolean = true) extends LeafNode {
  require(allowTempView || allowPermanentView)
  override lazy val resolved: Boolean = false
  override def output: Seq[Attribute] = Nil
}
```
such that the exception message can be updated to
`Table or temporary view not found for 'DROP TABLE': t`.

One downside is that if `t` is resolved to a view, the message will become:
```
t is a permanent view. 'DROP TABLE' expects a table or temporary view.
```
instead of (current message):
```
Cannot drop a view with DROP TABLE. Please use DROP VIEW instead
```
, meaning it will lose the "hint" (which we can add it to `Unresolved*` if required).

@cloud-fan WDYT? If you are OK with introducing `allowPermanentView`, I can do it as a separate PR.

Reply:
Sorry I don't get it. In this case, `testcat.db.notbl` is not a permanent view, but a non-existing relation.

Reply:
When I read `Table or view not found for 'DROP TABLE': t`, it sounded like `DROP TABLE` also supports permanent views.

So, I was suggesting `Table or temporary view not found for 'DROP TABLE': t` since `DROP TABLE` requires either a table or temporary view. But, I guess `Table or view not found` sounds better if we want to point out the non-existing relation.

Reply:
> since DROP TABLE requires either a table or temporary view

Do we have such a test case for v2 DROP TABLE?

Reply:
v2 `DROP TABLE` only handles `ResolvedTable`. If it's resolved to a view, it will be handled in `ResolveSessionCatalog` and tested here:
https://github.com/apache/spark/blob/fdd6c73b3cfac5af30c789c7f70b92367a79f7e7/sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala#L1056-L1070